### PR TITLE
[LibFix] Fix get_service_id to handle multiple instances of service

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Sample output for osd:

ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.10.service                                                loaded active running Ceph osd.10 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.13.service                                                loaded active running Ceph osd.13 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.16.service                                                loaded active running Ceph osd.16 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.2.service                                                 loaded active running Ceph osd.2 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.4.service                                                 loaded active running Ceph osd.4 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.7.service                                                 loaded active running Ceph osd.7 for f4ae9d84-e2a9-11ed-bb09-fa163e695471
  
  
 new output = >
 ['ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.10.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.13.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.16.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.2.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.4.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.7.service', '  ']
 
 To remove the trailing empty entry, added [:1] for return
 
 Final output =>['ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.1.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.11.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.14.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.17.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.5.service', 'ceph-f4ae9d84-e2a9-11ed-bb09-fa163e695471@osd.8.service']

**Comment by vinpapnoi** [comment]
LGTM

**Review by sayaleeraut — CHANGES_REQUESTED** [review]
LGTM

**Review by sayaleeraut — CHANGES_REQUESTED** [review]

File: cli/utilities/utils.py [hunk 1]
```diff
@@ -415,6 +422,21 @@ def set_service_state(node, service_id, state):
...
 
 
+def get_service_state(node, service):
+    """
+    Returns the state of a given service
+    Args:
+        node (ceph): Node to execute the cmd
+        service: Service id / name
+
+    Returns (str): Service ID
```
[the remark this answers is not in the document]
Comment (pranavprakash20) [reply cli/utilities/utils.py:432]: @sayaleeraut Noted, will update the same

[the remark this answers is not in the document]
Comment (pranavprakash20) [reply cli/utilities/utils.py]: Done

**Comment by pranavprakash20** [comment]
Changes has been made to return just the state of the service rather than the whole output of status command. Verified it with a sample test. and results can be seen here : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1KHRRU/

**Review by sayaleeraut — APPROVED** [review]
LGTM

**Comment by pranavprakash20** [comment]
run result: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DEVJ2O/

**Comment by pranavprakash20** [comment]
run results for the test fixes : https://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-2O7R9V

The cluster health warning error can be ignored as the above was tested on an existing cluster and not a cluster specific for mgr. The get_service_id method is working as expected with the fix

File: cli/utilities/utils.py [hunk 2]
```diff
@@ -389,14 +389,20 @@ def get_service_id(node, service_name):
...
+        cmd=f"systemctl --type=service | grep ceph-{service_name}"
+    )
     if err:
         return None
-    return out.split(" ")[0]
+
+    service_ids = []
+    for item in out.split("\n"):
+        service_ids.append(item.strip().split(" ")[0])
+    return service_ids[:-1]  # [:-1] To ignore the trailing empty entry
```
Comment (vamahaja) [inline cli/utilities/utils.py:405]: Will there always be a empty string entry at the end?

Can u try with to strip the out value and then do split ?

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
Changes made. 
Verified : run log => http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-T2UD1M/

File: cli/utilities/utils.py (hunk 2 above)
Comment (pranavprakash20) [reply cli/utilities/utils.py]: Noted, Strip worked.  Updated

**Review by vamahaja — COMMENTED** [review]
LGTM

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**